ClientServerCommFormats.md

```diff
@@ -0,0 +1,478 @@
+# Client-Server Communication Formats
+## Version 1
+
+This file will specify the standard format for data exchanged and stored between systems of the software. These formats will be used in the following system interactions:
+
+* Clients to Server
+* Server to Clients
+
+The format for all data to be exchanged will be that of JSON objects (https://www.json.org/). This means that whenever clients and the server exchange information a JSON object will be sent with the required information. 
+
+This document will be updated throughout the duration of the project as needed. Objects will be removed and added over the course of development.
+
+There will be four types of objects sent between clients and the server. These will be `Update`, `Action`, `ServerError`, and `ClientError`. Each of the four objects will have different purposes depending on which properties of the object are filled in or what values are filled in for those properties. 
+
+Clients will send `Action` and `ClientError` objects.
+The server will send `Update` and `ServerError` objects.
+
+An `Action` object will be sent by a client whenever a player wants to perform an action such as registering, logging in, inviting another player, moving a piece on the board, etc...
+
+A `ClientError` object will be sent by a client whenever there is an error on the client side.
+
+An `Update` object will be sent by the server whenever the server is responding to a client's action. These are things like updating the game board, sending the client their acount information, etc...
+
+A  `ServerError` object will be sent by the server whenever there is an error on the server side. These are things like the player attempted to perform an invalid set of moves, the login information was incorrect, etc...
+
+Here is the structure for each of the four objects:
+
+## Action
+
+```javascript
+{
+  "objectType": "Action",
+  "communicationType": "requestMove",
+  "communicationVersion": 1,
+  "matchID": "",
+  "playerName": "",
+  "pieceID": 3,
+  "desiredMoves": [][],
+  "userName": "",
+  "userPassword": "",
+  "userEmail": "",
+  "playerOneName": "",
+  "playerTwoName": "",
+  "invitationFrom": "",
+  "invitationTo": "",
+  "invitationTime": "",
+  "playerQuitting": ""
+}
+```
+## ClientError 
```

Comment:
Do you have an example of where a ClientError would be sent? I can't think of a use case, since all communication from the client is going to be triggered by the user.